service/ztests/query-describe.yaml

```diff
@@ -0,0 +1,76 @@
+script: |
+  source service.sh
+  zed create -q test1
+  zed create -q test2
+  for file in multifrom.zed agg.zed; do
+    echo // === $file ===
+    query="$(cat $file | jq -Rsa .)"
+    curl -H "Accept: application/json" -d "{\"query\":$query,\"head\":{\"pool\":\"test1\"}}" $ZED_LAKE/query/describe |
+      zq -J 'sources := (over sources | id := "XXX")' -
+  done
+
+
+inputs:
+  - name: service.sh
+  - name: multifrom.zed
+    data: |
+      from (
+        pool test1
+        pool test2
+      ) | put foo := "bar"
+  - name: agg.zed
+    data: |
+      count() by key1:=v1, key2
+
+outputs:
+  - name: stdout
+    data: |
+      // === multifrom.zed ===
+      {
+          "sources": [
+              {
+                  "kind": "Pool",
+                  "name": "test1",
+                  "id": "XXX"
+              },
+              {
+                  "kind": "Pool",
+                  "name": "test2",
+                  "id": "XXX"
+              }
+          ],
+          "aggregations": [
+              null
+          ],
```

Comment:
Since there's no aggregation in this query, I'd expect `null` or `[]` here. Is there a reason `[null]` makes sense?

Reply:
`null` let's you know that for this particular leg of the flowgraph there is no aggregation.

Reply:
Then the `aggregations` and `sort_keys` arrays each have one entry per channel? That isn't obvious from the structure here. How about making it clear with something like this?

```json
      {
          "sources": [...],
          "channels": [
              {
                    "aggregation_keys": [...],
                    "sort_keys": [...]
              }
          ]
      }
```

Reply:
One thing this approach doesn't support is letting you know if there is an aggregation in the channel. We could add a `has_aggregation` field or keep the aggregation structure, any preference?

Reply:
I think it does support that. Each entry in the `channels` array is a record containing the aggregation keys and sort keys for the channel. If `aggregation_keys` is `null`, the channel has no aggregation. Is that sufficient.

Reply:
So `[]` means has aggregation but no keys whereas `null` means no aggregation? I suppose this works since an empty array in javascript evaluations as truthy.